Index: script.c
===================================================================
--- script.c	(revision 529)
+++ script.c	(working copy)
@@ -9306,7 +9322,12 @@
 			case MF_NORETURN:			map[m].flag.noreturn=1; break;
 			case MF_NOWARPTO:			map[m].flag.nowarpto=1; break;
 			case MF_NIGHTMAREDROP:		map[m].flag.pvp_nightmaredrop=1; break;
-			case MF_RESTRICTED:			map[m].flag.restricted=1; break;
+			case MF_RESTRICTED:
+				map[m].zone |= 1<<((int)atoi(val)+1);
+				if (map[m].zone != 0){
+					map[m].flag.restricted=1;
+				}
+				break;
 			case MF_NOCOMMAND:			map[m].nocommand = (!val || atoi(val) <= 0) ? 100 : atoi(val); break;
 			case MF_JEXP:				map[m].jexp = (!val || atoi(val) < 0) ? 100 : atoi(val); break;
 			case MF_BEXP:				map[m].bexp = (!val || atoi(val) < 0) ? 100 : atoi(val); break;
@@ -9331,9 +9352,15 @@
 {
 	int m,i;
 	const char *str;
+	const char *val=NULL;
 
 	str=script_getstr(st,2);
 	i=script_getnum(st,3);
+
+	if(script_hasdata(st,4)){
+		val = script_getstr(st,4);
+	}
+
 	m = map_mapname2mapid(str);
 	if(m >= 0) {
 		switch(i) {
@@ -9373,7 +9400,12 @@
 			case MF_NORETURN:			map[m].flag.noreturn=0; break;
 			case MF_NOWARPTO:			map[m].flag.nowarpto=0; break;
 			case MF_NIGHTMAREDROP:		map[m].flag.pvp_nightmaredrop=0; break;
-			case MF_RESTRICTED:			map[m].flag.restricted=0; break;
+			case MF_RESTRICTED:
+				map[m].zone ^= 1<<((int)atoi(val)+1);
+				if (map[m].zone == 0){
+					map[m].flag.restricted=0;
+				}
+				break;
 			case MF_NOCOMMAND:			map[m].nocommand=0; break;
 			case MF_JEXP:				map[m].jexp=100; break;
 			case MF_BEXP:				map[m].bexp=100; break;
@@ -14630,7 +14662,7 @@
 	BUILDIN_DEF(setmapflagnosave,"ssii"),
 	BUILDIN_DEF(getmapflag,"si"),
 	BUILDIN_DEF(setmapflag,"si*"),
-	BUILDIN_DEF(removemapflag,"si"),
+	BUILDIN_DEF(removemapflag,"si?"),
 	BUILDIN_DEF(pvpon,"s"),
 	BUILDIN_DEF(pvpoff,"s"),
 	BUILDIN_DEF(gvgon,"s"),
