Index: db/const.txt
===================================================================
--- db/const.txt	(revision 612)
+++ db/const.txt	(working copy)
@@ -1174,6 +1174,7 @@
 SC_BANANA_BOMB_SITDOWN	554
 SC_STOMACHACHE	555
 SC_MYSTERIOUS_POWDER	556
+SC_ALL_RIDING	557
 
 e_gasp	0
 e_what	1
Index: db/item_db2.txt
===================================================================
--- db/item_db2.txt	(revision 612)
+++ db/item_db2.txt	(working copy)
@@ -473,3 +473,4 @@
 // Purchase Shop Items
 //6377,Buy_Stall_Permit,Buy Stall Permit,3,200,,10,,,,,,,,,,,,,{},{},{}
 //12548,Buy_Shabby_Stall_Permit,Buy Shabby Stall Permit,11,500,,10,,,,,0xFFFFFFFF,8,2,,,,,,{ itemskill "ALL_BUYING_STORE",1; },{},{}
+12622,Reins_Of_Mounts,Reins Of Mounts,2,0,,10,,,,,0xFFFFFFFF,7,2,,,,,,{ sc_start SC_ALL_RIDING,9999,0; getitem 12622,1; },{},{}
Index: src/map/clif.c
===================================================================
--- src/map/clif.c	(revision 612)
+++ src/map/clif.c	(working copy)
@@ -9822,7 +9822,8 @@
 			sd->sc.data[SC_DEATHBOUND] ||
 			sd->sc.data[SC_BLADESTOP] ||
 			(sd->sc.data[SC_NOCHAT] && sd->sc.data[SC_NOCHAT]->val1&MANNER_NOITEM) ||
-			sd->sc.data[SC_CURSEDCIRCLE_TARGET]
+			sd->sc.data[SC_CURSEDCIRCLE_TARGET] ||
+			sd->sc.data[SC_ALL_RIDING]
 		))
 			break;
 
Index: src/map/pc.c
===================================================================
--- src/map/pc.c	(revision 612)
+++ src/map/pc.c	(working copy)
@@ -7417,7 +7417,7 @@
 {
 	int option = 0, skillnum = 0;
 
-	if( sd->sc.data[SC__GROOMY] )
+	if( sd->sc.data[SC__GROOMY] || sd->sc.data[SC_ALL_RIDING] )
 		return 0;
 	
 	if( (sd->class_&MAPID_UPPERMASK) == MAPID_KNIGHT || (sd->class_&MAPID_UPPERMASK) == MAPID_CRUSADER )
Index: src/map/status.c
===================================================================
--- src/map/status.c	(revision 612)
+++ src/map/status.c	(working copy)
@@ -740,6 +740,8 @@
 	StatusIconChangeTable[SC_LIFE_FORCE_F] |= SI_LIFE_FORCE_F;
 	StatusIconChangeTable[SC_STOMACHACHE] = SI_STOMACHACHE;
 	StatusIconChangeTable[SC_MYSTERIOUS_POWDER] = SI_MYSTERIOUS_POWDER;
+	
+	StatusIconChangeTable[SC_ALL_RIDING] = SI_ALL_RIDING;
 
 	//Other SC which are not necessarily associated to skills.
 	StatusChangeFlagTable[SC_ASPDPOTION0] = SCB_ASPD;
@@ -823,6 +825,8 @@
 	StatusChangeFlagTable[SC_BANANA_BOMB] = SCB_LUK;
 	StatusChangeFlagTable[SC_STOMACHACHE] |= SCB_STR|SCB_AGI|SCB_VIT|SCB_INT|SCB_DEX|SCB_LUK;
 	StatusChangeFlagTable[SC_MYSTERIOUS_POWDER] |= SCB_MAXHP;
+	
+	StatusChangeFlagTable[SC_ALL_RIDING] = SCB_SPEED;
 
 	if( !battle_config.display_hallucination ) //Disable Hallucination.
 		StatusIconChangeTable[SC_HALLUCINATION] = SI_BLANK;
@@ -1424,8 +1428,9 @@
 				sc->data[SC__IGNORANCE] || // Target afflicted with this debuff cannot use skills or magic.
 				sc->data[SC_DEEPSLEEP] ||
 				sc->data[SC_SATURDAYNIGHTFEVER] ||
-				sc->data[SC_CURSEDCIRCLE_TARGET]
-			))
+				sc->data[SC_CURSEDCIRCLE_TARGET] ||
+				sc->data[SC_ALL_RIDING]
+				))
 				return 0;
 
 			//Skill blocking.
@@ -4665,6 +4670,9 @@
 				val = max( val, 25 );
 			if( sd && sd->speed_rate + sd->speed_add_rate < 0 ) // permanent item-based speedup
 				val = max( val, -(sd->speed_rate + sd->speed_add_rate) );
+				
+			if( sc->data[SC_ALL_RIDING])
+				val = max( val, 25 );
 
 			speed_rate -= val;
 		}
@@ -7750,6 +7758,16 @@
 			val4 = tick / 10000;
 			tick = 10000;
 			break;
+			
+		case SC_ALL_RIDING:
+			if ( sc->data[SC_ALL_RIDING] )
+			{
+				status_change_end(bl, SC_ALL_RIDING, -1);
+				return 0;
+			}
+			if (pc_isriding(sd,OPTION_RIDING|OPTION_RIDING_DRAGON|OPTION_RIDING_WUG|OPTION_MADO))
+				return 0;
+			break;
 
 		default:
 			if( calc_flag == SCB_NONE && StatusSkillChangeTable[type] == 0 && StatusIconChangeTable[type] == 0 )
Index: src/map/status.h
===================================================================
--- src/map/status.h	(revision 612)
+++ src/map/status.h	(working copy)
@@ -541,6 +541,8 @@
 	SC_BANANA_BOMB_SITDOWN,
 	SC_STOMACHACHE,
 	SC_MYSTERIOUS_POWDER,
+	
+	SC_ALL_RIDING,
 
 	SC_MAX, //Automatically updated max, used in for's to check we are within bounds.
 } sc_type;
@@ -1161,7 +1163,7 @@
 //	SI_SPIRITS_SAVEINFO2 = 610,
 //	SI_MAGIC_CANDY = 611,
 //	SI_ = 612,
-//	SI_ALL_RIDING = 613,
+	SI_ALL_RIDING = 613,
 //	SI_ = 614,
 //	SI_MACRO = 615,
 //	SI_MACRO_POSTDELAY = 616,
