Index: trunk/rewrite/db/const.txt
===================================================================
--- trunk/rewrite/db/const.txt	(revision 615)
+++ trunk/rewrite/db/const.txt	(working copy)
@@ -1174,6 +1174,7 @@
 SC_BANANA_BOMB_SITDOWN	554
 SC_STOMACHACHE	555
 SC_MYSTERIOUS_POWDER	556
+SC_ALL_RIDING	557
 
 e_gasp	0
 e_what	1
Index: trunk/rewrite/db/item_db2.txt
===================================================================
--- trunk/rewrite/db/item_db2.txt	(revision 615)
+++ trunk/rewrite/db/item_db2.txt	(working copy)
@@ -473,3 +473,5 @@
 // Purchase Shop Items
 //6377,Buy_Stall_Permit,Buy Stall Permit,3,200,,10,,,,,,,,,,,,,{},{},{}
 //12548,Buy_Shabby_Stall_Permit,Buy Shabby Stall Permit,11,500,,10,,,,,0xFFFFFFFF,8,2,,,,,,{ itemskill "ALL_BUYING_STORE",1; },{},{}
+12622,Reins_Of_Mount,Reins Of Mount,11,0,,10,,,,,0xFFFFFFFF,7,2,,,,,,{ itemskill "ALL_RIDING",1; },{},{}
+
Index: trunk/rewrite/db/skill_cast_db.txt
===================================================================
--- trunk/rewrite/db/skill_cast_db.txt	(revision 615)
+++ trunk/rewrite/db/skill_cast_db.txt	(working copy)
@@ -1553,6 +1553,9 @@
 
 //-- ALL_GUARDIAN_RECALL
 2536,3000,0,300000,0,0,0
+
+//-- ALL_RIDING
+2538,2000,0,1000,0,-1,0
 //==========================================
 
 //===== Homunculus Skills ==================
Index: trunk/rewrite/db/skill_castnodex_db.txt
===================================================================
--- trunk/rewrite/db/skill_castnodex_db.txt	(revision 615)
+++ trunk/rewrite/db/skill_castnodex_db.txt	(working copy)
@@ -36,6 +36,7 @@
 512,3	//GS_TRACKING
 1014,1	//PR_REDEMPTIO
 2316,1	//LG_EXCEEDBREAK
+2538,7	//ALL_RIDING
 10010,3	//GD_BATTLEORDER
 10011,3	//GD_REGENERATION
 10012,6	//GD_RESTORE
Index: trunk/rewrite/db/skill_db.txt
===================================================================
--- trunk/rewrite/db/skill_db.txt	(revision 615)
+++ trunk/rewrite/db/skill_db.txt	(working copy)
@@ -1210,6 +1210,7 @@
 2535,0,0,4,0,0x1,0,1,0,no,0,0x1,0,none,0,	ALL_BUYING_STORE,Open Buying Store
 2536,0,6,4,0,0x1,0,1,1,yes,0,0,0,none,0,	ALL_GUARDIAN_RECALL,Guardian's Recall
 2537,9,6,16,0,0x1,0,2,1,yes,0,0,0,magic,0,	ALL_ODINS_POWER,Odin's Power
+2538,0,6,4,0,0x01,0,1,1,yes,0,0,0,none,0,	ALL_RIDING,Call Mount
 
 // Homunculus Skills
 8001,9,6,4,0,0x1,0,5,1,no,0,0,0,magic,0,         HLIF_HEAL,Healing Touch
Index: trunk/rewrite/db/skill_require_db.txt
===================================================================
--- trunk/rewrite/db/skill_require_db.txt	(revision 615)
+++ trunk/rewrite/db/skill_require_db.txt	(working copy)
@@ -968,6 +968,7 @@
 2535,0,0,1,0,0,0,99,0,0,none,0,6377,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//ALL_BUYING_STORE#Open Buying Store#
 2536,0,0,10,0,0,0,99,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//ALL_GUARDIAN_RECALL#Guardian's Recall#
 2537,0,0,20,0,0,0,99,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//ALL_ODINS_POWER#Odin's Power#
+2538,0,0,1,0,0,0,99,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//ALL_RIDING#Call Mount#
 
 // Homunculus Skills
 8001,0,0,13:16:19:22:25,0,0,0,99,0,0,none,0,545,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//HLIF_HEAL
Index: trunk/rewrite/src/map/skill.c
===================================================================
--- trunk/rewrite/src/map/skill.c	(revision 615)
+++ trunk/rewrite/src/map/skill.c	(working copy)
@@ -5574,6 +5574,7 @@
 		break;
 	case TF_HIDING:
 	case ST_CHASEWALK:
+	case ALL_RIDING:
 		if (tsce)
 		{
 			clif_skill_nodamage(src,bl,skillid,-1,status_change_end(bl, type, INVALID_TIMER)); //Hide skill-scream animation.
@@ -11851,7 +11852,7 @@
 	case BS_MAXIMIZE:		case NV_TRICKDEAD:	case TF_HIDING:			case AS_CLOAKING:		case CR_AUTOGUARD:
 	case ML_AUTOGUARD:		case CR_DEFENDER:	case ML_DEFENDER:		case ST_CHASEWALK:		case PA_GOSPEL:
 	case CR_SHRINK:			case TK_RUN:		case GS_GATLINGFEVER:	case TK_READYCOUNTER:	case TK_READYDOWN:
-	case TK_READYSTORM:		case TK_READYTURN:	case SG_FUSION:
+	case TK_READYSTORM:		case TK_READYTURN:	case SG_FUSION:			case ALL_RIDING:
 		if( sc && sc->data[status_skill2sc(skill)] )
 			return 1;
 	}
@@ -12785,7 +12786,7 @@
 	case BS_MAXIMIZE:		case NV_TRICKDEAD:	case TF_HIDING:			case AS_CLOAKING:		case CR_AUTOGUARD:
 	case ML_AUTOGUARD:		case CR_DEFENDER:	case ML_DEFENDER:		case ST_CHASEWALK:		case PA_GOSPEL:
 	case CR_SHRINK:			case TK_RUN:		case GS_GATLINGFEVER:	case TK_READYCOUNTER:	case TK_READYDOWN:
-	case TK_READYSTORM:		case TK_READYTURN:	case SG_FUSION:
+	case TK_READYSTORM:		case TK_READYTURN:	case SG_FUSION:			case ALL_RIDING:
 		if( sc && sc->data[status_skill2sc(skill)] )
 			return req;
 	}
Index: trunk/rewrite/src/map/skill.h
===================================================================
--- trunk/rewrite/src/map/skill.h	(revision 615)
+++ trunk/rewrite/src/map/skill.h	(working copy)
@@ -1486,6 +1486,7 @@
 	ALL_BUYING_STORE,
 	ALL_GUARDIAN_RECALL,
 	ALL_ODINS_POWER,
+	ALL_RIDING,
 
 	HLIF_HEAL = 8001,
 	HLIF_AVOID,
Index: trunk/rewrite/src/map/status.c
===================================================================
--- trunk/rewrite/src/map/status.c	(revision 615)
+++ trunk/rewrite/src/map/status.c	(working copy)
@@ -540,6 +540,8 @@
 	set_sc( GN_FIRE_EXPANSION_TEAR_GAS    , SC_TEARGAS     , SI_FIRE_EXPANSION_TEAR_GAS    , SCB_NONE );
 	set_sc( GN_MANDRAGORA                 , SC_MANDRAGORA  , SI_MANDRAGORA                 , SCB_INT );
 
+	set_sc( ALL_RIDING           , SC_ALL_RIDING      , SI_ALL_RIDING       , SCB_SPEED );
+
 	set_sc( HLIF_AVOID           , SC_AVOID           , SI_BLANK           , SCB_SPEED );
 	set_sc( HLIF_CHANGE          , SC_CHANGE          , SI_BLANK           , SCB_VIT|SCB_INT );
 	set_sc( HFLI_FLEET           , SC_FLEET           , SI_BLANK           , SCB_ASPD|SCB_BATK|SCB_WATK );
@@ -1424,7 +1426,8 @@
 				sc->data[SC__IGNORANCE] || // Target afflicted with this debuff cannot use skills or magic.
 				sc->data[SC_DEEPSLEEP] ||
 				sc->data[SC_SATURDAYNIGHTFEVER] ||
-				sc->data[SC_CURSEDCIRCLE_TARGET]
+				sc->data[SC_CURSEDCIRCLE_TARGET] ||
+				sc->data[SC_ALL_RIDING]
 			))
 				return 0;
 
@@ -4659,6 +4662,8 @@
 				val = max( val, sc->data[SC_GT_REVITALIZE]->val2 );
 			if( sc->data[SC_WIND_STEP_OPTION] )
 				val = max( val, sc->data[SC_WIND_STEP_OPTION]->val2 );
+			if( sc->data[SC_ALL_RIDING] )
+				val = max( val, 25 );
 
 			//FIXME: official items use a single bonus for this [ultramage]
 			if( sc->data[SC_SPEEDUP0] ) // temporary item-based speedup
@@ -7940,6 +7945,10 @@
 //			sc->opt3 |= OPT3_CONTRACT;
 //			opt_flag = 0;
 //			break;
+		case SC_ALL_RIDING:
+			sc->opt3 |= OPT3_ALL_RIDING;
+			opt_flag = 0;
+			break;
 		//OPTION
 		case SC_HIDING:
 			sc->option |= OPTION_HIDE;
@@ -8119,6 +8128,7 @@
 		case SC_FOOD_DEX_CASH:
 		case SC_FOOD_INT_CASH:
 		case SC_FOOD_LUK_CASH:
+		case SC_ALL_RIDING:
 			continue;
 		}
 
@@ -8719,6 +8729,10 @@
 //		sc->opt3 &= ~OPT3_CONTRACT;
 //		opt_flag = 0;
 //		break;
+	case SC_ALL_RIDING:
+		sc->opt3 &= ~OPT3_ALL_RIDING;
+		opt_flag = 0;
+		break;
 	default:
 		opt_flag = 0;
 	}
Index: trunk/rewrite/src/map/status.h
===================================================================
--- trunk/rewrite/src/map/status.h	(revision 615)
+++ trunk/rewrite/src/map/status.h	(working copy)
@@ -542,6 +542,8 @@
 	SC_STOMACHACHE,
 	SC_MYSTERIOUS_POWDER,
 
+	SC_ALL_RIDING,
+
 	SC_MAX, //Automatically updated max, used in for's to check we are within bounds.
 } sc_type;
 
@@ -1161,7 +1163,7 @@
 //	SI_SPIRITS_SAVEINFO2 = 610,
 //	SI_MAGIC_CANDY = 611,
 //	SI_ = 612,
-//	SI_ALL_RIDING = 613,
+	SI_ALL_RIDING = 613,
 //	SI_ = 614,
 //	SI_MACRO = 615,
 //	SI_MACRO_POSTDELAY = 616,
@@ -1275,6 +1277,7 @@
 	OPT3_SOULLINK         = 0x00008000,
 	OPT3_UNDEAD           = 0x00010000,
 	OPT3_CONTRACT         = 0x00020000,
+	OPT3_ALL_RIDING       = 0x00040000,
 };
 
 enum {
